### ISSUE TITLE
Improve invalid deferred module definition Error reporting

### ISSUE THREAD

**Issue by Ahathaway97** [IntrospectiveSystems/xGraph#65]
When a deferred module is improperly formatted (like below, taken from testing the BankAccount and BankPatron test system) it throws nondescript unhandled promise rejection errors. 

Proposed solution:
- have it throw an error noting the improperly formatted object and skip it to attempt running without the object.

```
{
	"Sources": {
		"core":	"mb://modulebroker.xgraphdev.com"
	},
	"Modules": {
		"BankAccount":{
			"Module": "xGraph.BankAccount", 
			"Source": "core",
			"Par": {
			}
		},
		"BankPatron":{
			"Module": "xGraph.BankPatron", 
			"Source": "core",
			"Par": {
			}
		},
		"Deferred": [
			{}
		]
	}
}
```

**Comment by valyrie97** [comment]
Implementation something along the lines of, test for the keys inside the deferred object first, then access them. If the format doesnt make sense, inform the user by dumping the object and throwing E_INVALID_FORMAT

**Comment by tjgionet** [comment]
Fixed by commit c7808ddd249964f7e0220560338b34b9b2a51aa8
